### PR TITLE
docs: add editable installation instructions

### PR DESCRIPTION
**Pull request by oscarandersson8218**
Add instructions on how to install ExecuTorch as an editable installation.

### CODE REVIEW

File: docs/source/getting-started-setup.md [hunk 1]
```diff
@@ -129,6 +129,11 @@ Follow these steps:
    ./install_requirements.sh
    ```
 
+   **Optional:** Install ExecuTorch as an editable installation:
```
Comment (digantdesai) [inline docs/source/getting-started-setup.md:132]: so is it a two step process?
i.e. 1. run install_requirements.sh, and 2. do another pip install with these flags?

Comment (oscarandersson8218) [reply docs/source/getting-started-setup.md:132]: Yes, or you could replace:
`pip install . --no-build-isolation` in install_requirements.sh with:
`pip install --editable . --config-settings editable_mode=strict --no-build-isolation`
and then run install_requirements.sh.

Comment (digantdesai) [reply docs/source/getting-started-setup.md:132]: yeah if its purely additive step after `install_requirements.sh` then its the cleanest IMHO,

**Comment by digantdesai** [comment]
cc @guangy10 @dbort - any issues with this as a general recommendation for devs?

**Pull request closed without merging**